WcaOnRails/app/models/role.rb

```diff
@@ -2,4 +2,24 @@
 
 class Role < ApplicationRecord
   belongs_to :user
+  belongs_to :group, class_name: "UserGroup"
+  belongs_to :metadata, polymorphic: true, optional: true
+
+  def group_type
```

Comment:
Neat trick: Rails can do

`delegate :group_type, to: :group`

which you can use to completely get rid of this def